src/coreclr/vm/codeman.cpp

```diff
@@ -1514,64 +1543,50 @@ void EEJitManager::SetCpuInfo()
                                     {
                                         __cpuidex(cpuidInfo, 0x00000007, 0x00000000);
 
-                                        if ((cpuidInfo[CPUID_EBX] & (1 << 5)) != 0)                               // AVX2
+                                        if ((cpuidInfo[CPUID_EBX] & (1 << 5)) != 0)                                     // AVX2
                                         {
                                             CPUCompileFlags.Set(InstructionSet_AVX2);
 
-                                            if (DoesOSSupportAVX512() && (avx512StateSupport() == 1))      // XGETBV XRC0[7:5] == 111
+                                            if (DoesOSSupportAVX512() && (avx512StateSupport() == 1))                   // XGETBV XRC0[7:5] == 111
                                             {
-                                                if ((cpuidInfo[CPUID_EBX] & (1 << 16)) != 0)                     // AVX512F
+                                                // While the AVX-512 ISAs can be individually lit-up, they really
```

Comment:
Should this simplifying assumption be rather encoded as instruction set implication in https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt#L15C1-L16 so that it is indentical between the runtime JIT and AOT compilers?

Reply:
I don't think we have a way to specify such multiple dependencies today. We likewise can't model that `Avx512F` implies both `Avx2` and `Fma` (and the unimplemented `F16C`) itself.

We could potentially model it as "just" `Avx512` in the `CORINFO_InstructionSet`, but that may be "limiting" if we ever need to change it in the future and wouldn't map cleanly to the way we expose the managed APIs or how we currently map managed APIs back to the tracked `InstructionSet` enum in JIT/AOT.

It would be nice if we could more easily model such dependencies (including circular ones), however.

Reply:
You can always extend what the generator of instruction set definitions does.

These implications belong to the auto-generated methods like `EnsureValidInstructionSetSupport`.

Reply:
Right. Just there is significant complexity in doing so.

I'll see how feasible it is to update the bits here.